### PR TITLE
util: support clustered index for table scan

### PR DESCRIPTION
**Pull request by wshwsh12**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

Proposal: [Clustered Index](https://github.com/pingcap/tidb/blob/master/docs/design/2020-05-08-cluster-index.md) 

What's Changed:

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test https://github.com/ngaut/unistore/pull/394
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 
- No Need.

### CODE REVIEW

File: tablecodec/tablecodec.go [hunk 1]
```diff
@@ -244,11 +244,14 @@ func DecodeTableID(key kv.Key) int64 {
 
 // DecodeRowKey decodes the key and gets the handle.
 func DecodeRowKey(key kv.Key) (kv.Handle, error) {
-	if len(key) != RecordRowKeyLen || !hasTablePrefix(key) || !hasRecordPrefixSep(key[prefixLen-2:]) {
+	if len(key) < prefixLen || !hasTablePrefix(key) || !hasRecordPrefixSep(key[prefixLen-2:]) {
```
Comment (coocood) [inline tablecodec/tablecodec.go:247]: The minimum common handle length is 9, so if len(key) < RecordRowKeyLen, we need to return error.

Comment (wshwsh12) [reply tablecodec/tablecodec.go]: Ok.

File: util/rowcodec/decoder.go [hunk 2]
```diff
@@ -272,10 +272,19 @@ func (decoder *ChunkDecoder) tryAppendHandleColumn(colIdx int, col *ColInfo, han
...
 		return false
 	}
-	if col.ID == decoder.handleColIDs[0] {
+	if handle.IsInt() && col.ID == decoder.handleColIDs[0] {
 		chk.AppendInt64(colIdx, handle.IntValue())
 		return true
 	}
+	for i, id := range decoder.handleColIDs {
+		if col.ID == id {
+			_, val, err := codec.DecodeOne(handle.EncodedCol(i))
```
Comment (coocood) [inline util/rowcodec/decoder.go:281]: We need to use the `ColInfo` to decode the column,

Call `decodeColToChunk`.

Comment (coocood) [reply util/rowcodec/decoder.go:281]: Need to call this one https://github.com/pingcap/tidb/blob/master/util/codec/codec.go#L972

File: go.mod [hunk 3]
```diff
@@ -63,3 +64,5 @@ require (
 )
 
 go 1.13
+
+replace github.com/ngaut/unistore => github.com/wshwsh12/unistore v0.0.0-20200603025657-e850184de1f3
```
Comment (coocood) [inline go.mod:68]: Seems we don't need update unistore at this PR.

**Comment by coocood** [comment]
LGTM

**Pull request merged**